### PR TITLE
libpod: add hidden env to set sqlite timeout

### LINKED ISSUE #23236: RFE: expose sqliteOptionBusyTimeout as config or env variable.
### Feature request description

A clear and concise description of what the feature request is about.

We are executing several stress tests against podman with [engine-stressor](https://github.com/containers/engine-stressor) and depending of the tuning/settings in the system with stress-ng tool we start seeing the famous "**save transaction: database is locked**"

Would be nice to have a way to increase/decreate **sqliteOptionBusyTimeout** to see if that affects the system without the need to recompile the libpod. On top that, it worth mentioning that changing that value should not corrupt the database itself.

I could offer help to implement such mechanism but better listen the team how they would like to drive this request.

Reference:
https://github.com/containers/podman/blob/b3afb8fc39c8d2186474da79c3b88522ee251a22/libpod/sqlite_state.go#L45

Thanks
Douglas

### Suggest potential solution

Add such option into config?
Env variable? 

### Have you considered any alternatives?

Rebuild manually podman/libpod

### Additional context

Doing stress test against podman to a automobile certification which uses podman as container engine.

### PR DESCRIPTION
**Pull request by Luap99**
Some users want to experiment with different timeout values.

Fixes #23236

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: libpod/sqlite_state.go [hunk 1]
```diff
@@ -71,7 +67,17 @@ func NewSqliteState(runtime *Runtime) (_ State, defErr error) {
...
+	// Timeout is in ms, so set it to 100s to have enough time to retry the operations.
+	// Some users might want to experiment with different timeout values (#23236)
+	// DO NOT DOCUMENT or recommend PODMAN_SQLITE_BUSY_TIMEOUT outside of testing.
+	busyTimeout := "100000"
+	if env, ok := os.LookupEnv("PODMAN_SQLITE_BUSY_TIMEOUT"); ok {
+		busyTimeout = env
+	}
+	sqliteOptionBusyTimeout := "&_busy_timeout=" + busyTimeout
+
+	conn, err := sql.Open("sqlite3", filepath.Join(basePath, sqliteOptions+sqliteOptionBusyTimeout))
```
Comment (mheon) [inline libpod/sqlite_state.go:80]: Do we get a vaguely sane error if the user enters something that is not a number? I have visions of someone having this environment variable set to a garbage value and having a hell of a time debugging that.

Comment (Luap99) [reply libpod/sqlite_state.go:80]:
```
Error: beginning transaction: Invalid _busy_timeout: notInt: strconv.ParseInt: parsing "notInt": invalid syntax
```

Comment (mheon) [reply libpod/sqlite_state.go:80]: Perfect, thank god for SQLite's excellent error messages

**Comment by mheon** [comment]
LGTM

File: libpod/sqlite_state.go [hunk 2]
```diff
@@ -71,7 +67,17 @@ func NewSqliteState(runtime *Runtime) (_ State, defErr error) {
...
 	}
 
-	conn, err := sql.Open("sqlite3", filepath.Join(basePath, sqliteOptions))
+	// Make sure busy timeout is set to high value to keep retrying when the db is locked.
+	// Timeout is in ms, so set it to 100s to have enough time to retry the operations.
+	// Some users might want to experiment with different timeout values (#23236)
+	// DO NOT DOCUMENT or recommend PODMAN_SQLITE_BUSY_TIMEOUT outside of testing.
+	busyTimeout := "100000"
+	if env, ok := os.LookupEnv("PODMAN_SQLITE_BUSY_TIMEOUT"); ok {
+		busyTimeout = env
```
Comment (rhatdan) [inline libpod/sqlite_state.go:76]: Should we add a logrus.Debug here to remind users that the environment variable is set?

**Comment by rhatdan** [comment]
/hold
Asked question

File: libpod/sqlite_state.go (hunk 2 above)
Comment (Luap99) [reply libpod/sqlite_state.go:76]: yes good idea

**Pull request merged**